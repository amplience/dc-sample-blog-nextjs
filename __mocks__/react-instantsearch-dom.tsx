import React, { ReactElement } from 'react';

export const InstantSearch = (): ReactElement => <div>InstantSearch mock</div>;
