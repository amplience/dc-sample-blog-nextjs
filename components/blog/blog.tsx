import React, { ReactElement } from 'react';
import BlogPostAuthor from '../blog-post-author/blog-post-author.component';
import BlogPostHeroBanner from '../hero-banner/blog-post-hero-banner';
import Content from '../content/content';
import BlogPost from '../../common/interfaces/blog-post.interface';
import Picture from '../picture/picture';
import TagChips from '../tag-chips/tag-chips';

const Blog = ({ blogPost }: { blogPost: BlogPost }): ReactElement => {
  return (
    <>
      <div className="content-header">
        <BlogPostAuthor authors={blogPost.authors} date={blogPost.date} readTime={blogPost.readTime} />
        <TagChips tags={blogPost.tags} />
        <BlogPostHeroBanner title={blogPost.title} subTitle={blogPost.description} />
      </div>
      <div className="blog-image">
        <Picture
          image={blogPost.image}
          sources={[
            {
              di: { w: 2048, sm: 'c', scaleFit: 'poi' },
              media: '(min-width: 1080px)'
            },
            {
              di: { w: 1080, sm: 'c', scaleFit: 'poi' },
              media: '(min-width: 768px)'
            },
            {
              di: { w: 768, sm: 'c', scaleFit: 'poi' },
              media: '(min-width: 480px)'
            },
            {
              di: { w: 480, sm: 'c', scaleFit: 'poi' }
            }
          ]}
        />
      </div>
      <div className="content-wrapper">
        <Content content={blogPost.content} />
      </div>
      <style jsx>{`
        .content-header {
          margin: auto;
          max-width: 740px;
        }

        .content-header :global(div.tag-chips) {
          padding-top: 25px;
        }

        .content-wrapper {
          margin: auto;
          max-width: 675px;
          padding-bottom: 100px;
        }

        .blog-image :global(img) {
          object-fit: cover;
          width: 100%;
        }

        :global(footer) {
          margin-top: 100px;
        }

        @media (max-width: 675px) {
          .content-header,
          .content-footer,
          .content-wrapper {
            display: block;
            padding: 0 20px;
          }
        }
      `}</style>
    </>
  );
};

export default Blog;
