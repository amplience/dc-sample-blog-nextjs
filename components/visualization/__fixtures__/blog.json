{
  "_meta": {
    "schema": "https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/blog.json",
    "deliveryId": "9cb44d61-d00b-49b7-b23b-72bb9a0a1de6",
    "deliveryKey": "blog",
    "name": "blog"
  },
  "title": "Blog",
  "heading": "heading",
  "searchPlaceHolder": "text"
}
