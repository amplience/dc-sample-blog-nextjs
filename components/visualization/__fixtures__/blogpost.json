{
  "id": "8d6943c7-6028-4fac-b45e-57fc63bd032a",
  "title": "My First Bloblog Post",
  "date": "August 07, 2019",
  "description": "Some text for the description that explains this bloblog post",
  "_meta": {
    "schema": "https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/blog.json",
    "deliveryId": "9cb44d61-d00b-49b7-b23b-72bb9a0a1de6",
    "deliveryKey": "blog",
    "name": "blog"
  },
  "authors": [
    {
      "name": "Sir Bloblog",
      "_meta": {
        "schema": "https://schema.localhost.com/author.json",
        "deliveryId": "b1379f67-8934-49cb-94b3-58b679096c2c",
        "name": "bloblog-author"
      },
      "avatar": {
        "image": {
          "defaultHost": "1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io",
          "endpoint": "bloblogltd",
          "name": "man-business-suit",
          "id": "f9549d3d-f616-487c-b1ad-fad5152f2c7d",
          "_meta": {
            "schema": "http://bigcontent.io/cms/schema/v1/core#/definitions/image-link"
          },
          "mediaType": "i"
        },
        "altText": "The dashing Sir Bloblog avatar",
        "_meta": {
          "schema": "https://schema.localhost.com/image.json",
          "deliveryId": "9ef5a6d0-5a34-414f-9fd3-1098698c6629",
          "name": "bloblog-avatar"
        },
        "src": "//1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io/i/bloblogltd/man-business-suit"
      }
    }
  ],
  "readTime": 12,
  "image": {
    "image": {
      "defaultHost": "1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io",
      "endpoint": "bloblogltd",
      "name": "friends-shopping-bags",
      "id": "edc77f02-0e92-4b1a-88f9-711cacb5d650",
      "_meta": {
        "schema": "http://bigcontent.io/cms/schema/v1/core#/definitions/image-link"
      },
      "mediaType": "i"
    },
    "altText": "crotch bags",
    "_meta": {
      "schema": "https://schema.localhost.com/image.json",
      "deliveryId": "4b44e870-bf71-43f3-81b3-c46c32f4cde5",
      "name": "crotchy-bag-image"
    },
    "src": "//1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io/i/bloblogltd/friends-shopping-bags"
  },
  "content": [
    {
      "image": {
        "defaultHost": "1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io",
        "endpoint": "bloblogltd",
        "name": "casual-wear",
        "id": "e1b511d2-1a33-47e7-8dc7-f460534cb0c7",
        "_meta": {
          "schema": "http://bigcontent.io/cms/schema/v1/core#/definitions/image-link"
        },
        "mediaType": "i"
      },
      "altText": "Casual crotch shot",
      "_meta": {
        "schema": "https://schema.localhost.com/image.json",
        "deliveryId": "a61f458e-7c22-4b5b-9b3d-962df3cbcdcd",
        "name": "casual-crotch-shot"
      },
      "src": "//1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io/i/bloblogltd/casual-wear"
    },
    {
      "_meta": {
        "schema": "https://schema.localhost.com/text.json",
        "deliveryId": "f08432cd-525f-4e50-ab34-3316081f48bd",
        "name": "my-first-bloblog-text"
      },
      "text": "# This is my first Bloblog\n\n## Look-a-header 2\n\n### Look-a-header 3\n\n#### Look-a-header 4\n\n##### Look-a-header 5\n\n###### Look-a-header 6\n\n_**This**_ _is_ _some_ ~~sample~~ **text**.\n\n![alt text](https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png \"Logo Title Text 1\")\n\nThis is some more sample text with a [LINK](http://bbc.co.uk)\n\n> Comments are cool!\n> \n> See what I mean\n\nLook some code:\n\n```javascript\nconst something = 'anything';\ncalls(something);\n\n```\n\ninline `code` is awesome underlined\n\n* * *\n\n<h4>A title</h4>\n\nA table too:\n\n| Hello | world | table |\n| --- | --- | --- |\n| some | values | here |\n| some | values | here |\n| some | values | here |\n| some | values | here |\n\nGet listed\n\n*   Item One\n*   Item Two\n*   Item Three"
    },
    {
      "_meta": {
        "schema": "https://schema.localhost.com/video.json",
        "deliveryId": "ad568b60-fd02-4302-9ccc-9e35dac24255",
        "name": "my-first-bloblog-video"
      },
      "video": {
        "defaultHost": "1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io",
        "endpoint": "bloblogltd",
        "name": "SampleVideo_1280x720_5mb",
        "id": "721044de-d125-4a1a-8ddc-2201b9463f2d",
        "_meta": {
          "schema": "http://bigcontent.io/cms/schema/v1/core#/definitions/video-link"
        },
        "mediaType": "v"
      },
      "srcSet": [
        "https://1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io/v/bloblogltd/SampleVideo_1280x720_5mb/mp4_240p?protocol=https"
      ]
    }
  ],
  "tags": ["sir bloblog", "you're it", "No tiggy backs"]
}
