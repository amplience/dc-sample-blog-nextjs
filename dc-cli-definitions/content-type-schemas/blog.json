{
  "body": "../../schemas/blog.json",
  "schemaId": "https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/blog.json",
  "validationLevel": "CONTENT_TYPE"
}