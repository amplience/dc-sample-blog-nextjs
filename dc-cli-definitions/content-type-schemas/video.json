{
  "body": "../../schemas/video.json",
  "schemaId": "https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/video.json",
  "validationLevel": "CONTENT_TYPE"
}