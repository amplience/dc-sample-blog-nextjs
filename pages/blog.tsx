import { NextPage } from 'next';
import BlogPost from '../common/interfaces/blog-post.interface';
import Layout from '../layouts/default';
import getBlogPost from '../common/services/blog-post.service';
import Microdata from '../components/microdata/microdata';
import { NextSeo } from 'next-seo';
import Blog from '../components/blog/blog';
import SharePost from '../components/share-post/share-post';
import { Image } from 'dc-delivery-sdk-js';
import { defaultClientConfig } from '../common/services/dynamic-content-client-config';

interface BlogPostProps {
  blogPost: BlogPost;
}

const BlogPostPage: NextPage<BlogPostProps> = ({ blogPost }: BlogPostProps) => {
  const blogImage = new Image(blogPost.image.image, defaultClientConfig).url().build();

  const seoParams: { [key: string]: string | any } = {
    title: blogPost.title,
    description: blogPost.description,
    twitter: {
      cardType: 'summary_large_image'
    },
    openGraph: {
      title: blogPost.title,
      description: blogPost.description,
      images: [
        {
          url: `${blogImage}?w=1080`,
          width: 1080
        }
      ]
    }
  };

  if (process.env.ROBOTS_META_TAG_NOINDEX === 'true') {
    seoParams.noindex = true;
  }

  return (
    <Layout>
      <NextSeo {...seoParams} />
      <Blog blogPost={blogPost} />
      <div className="content-footer">
        <SharePost twitterText={blogPost.title} />
      </div>
      <Microdata
        description={blogPost.description}
        headline={blogPost.title}
        imageUrl={blogImage}
        authors={blogPost.authors}
        datePublished={blogPost.date}
      />
      <style jsx>{`
        .content-footer {
          margin: auto;
          max-width: 740px;
        }
      `}</style>
    </Layout>
  );
};

BlogPostPage.getInitialProps = async ({ query }) => {
  const { vse, blogId } = query;
  const stagingEnvironment = vse ? `//${vse.toString()}` : undefined;
  try {
    const blogPostId = blogId ? blogId.toString() : '';
    const blogPost = await getBlogPost(blogPostId, stagingEnvironment);
    return { blogPost };
  } catch (err) {
    throw err;
  }
};

export default BlogPostPage;
