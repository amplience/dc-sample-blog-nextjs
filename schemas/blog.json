{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "id": "https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/blog.json",
  "title": "Blog list",
  "description": "A list of blogs",
  "allOf": [
    {
      "$ref": "http://bigcontent.io/cms/schema/v1/core#/definitions/content"
    }
  ],
  "type": "object",
  "properties": {
    "title": {
      "title": "Title",
      "description": "Title of the blog",
      "type": "string",
      "minLength": 1,
      "maxLength": 100,
      "examples": ["Amplience Product Blog"]
    },
    "heading": {
      "title": "heading",
      "description": "Heading of the blog",
      "type": "string",
      "minLength": 1,
      "maxLength": 300,
      "examples": ["News & updates from the Amplience production development"]
    },
    "searchPlaceHolder": {
      "title": "Search Placeholder",
      "description": "Placeholder text for search box",
      "type": "string",
      "minLength": 0,
      "maxLength": 150,
      "examples": ["Search for articles and topics"]
    },
    "_meta":{
      "type":"object",
      "properties":{
        "deliveryKey":{
          "type":"string",
          "title":"Delivery key",
          "description": "The delivery key will be used to retrieve the blog when it is built",
          "examples": ["blog"]
        }
      },
      "required": ["deliveryKey"]
    }
  },
  "required": ["title", "heading", "searchPlaceHolder"],
  "propertyOrder": ["title", "heading", "searchPlaceHolder","_meta"]
}
