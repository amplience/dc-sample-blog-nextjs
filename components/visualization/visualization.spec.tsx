import waitUntil from 'async-wait-until';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { findResultsState } from 'react-instantsearch-dom/server';
import { Blog } from '../../common/interfaces/blog.interface';
import { VideoProfile } from '../../common/services/video.service';
import PageLoader from '../page-loader/page-loader';
import Visualization from './visualization';
import * as blogFixture from './__fixtures__/blog.json';
import * as blogPostFixture from './__fixtures__/blogpost.json';

const mockFetch = jest.fn();
const mockGetContentItemById = jest.fn();
jest.mock('search-insights');
jest.mock('../../common/services/vse.service', () => () => mockGetContentItemById());
jest.mock('react-instantsearch-dom', () => ({
  ...jest.requireActual('react-instantsearch-dom'),
  connectStateResults: templateFn => params => templateFn(params)
}));
jest.mock('algoliasearch', () => () => 'mockSearchClient');
jest.mock('react-instantsearch-dom/server', () => ({
  findResultsState: jest.fn()
}));
jest.mock('isomorphic-unfetch', () => {
  return (): jest.Mock => mockFetch();
});

describe('Visualization', (): void => {
  beforeEach(() => {
    process.env.ALGOLIA_APPLICATION_ID = 'algolia-app-id';
    process.env.SEARCH_API_KEY = 'algolia-search-key';
    process.env.SEARCH_INDEX_NAME_STAGING = 'algolia-index-name';

    mockFetch.mockImplementationOnce((): { json: () => VideoProfile; status: number } => {
      return {
        json(): VideoProfile {
          return {
            media: [{ src: 'https://i1-qa.adis.ws/v/bloblogltd/SampleVideo_1280x720_5mb/mp4_240p' }]
          };
        },
        status: 200
      };
    });
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  async function renderVisualization<T>(contentItem: T) {
    mockGetContentItemById.mockResolvedValue(contentItem);

    const wrapper = shallow<Visualization>(<Visualization stagingEnvironment="vse" contentItemId="content" />, {
      lifecycleExperimental: true
    });

    expect(toJson(wrapper)).toMatchSnapshot();

    await waitUntil(() => !wrapper.contains(<PageLoader />));

    expect(mockGetContentItemById).toHaveBeenCalled();
    expect(toJson(wrapper)).toMatchSnapshot();
  }

  it('should render text', async (): Promise<void> => {
    const contentItem = {
      text: '# Markdown Text'
    };
    await renderVisualization(contentItem);
  });

  it('should render image', async (): Promise<void> => {
    const contentItem = {
      altText: 'altText',
      image: {
        defaultHost: 'i1-qa.adis.ws',
        endpoint: 'bloblogltd',
        name: 'friends-shopping-bags',
        id: 'edc77f02-0e92-4b1a-88f9-711cacb5d650'
      }
    };

    await renderVisualization(contentItem);
  });

  it('should render video', async (): Promise<void> => {
    const contentItem = {
      video: {
        defaultHost: 'i1-qa.adis.ws',
        endpoint: 'bloblogltd',
        name: 'SampleVideo_1280x720_5mb',
        id: '721044de-d125-4a1a-8ddc-2201b9463f2d'
      }
    };

    await renderVisualization(contentItem);
  });

  it('should handle vse errors', async (): Promise<void> => {
    mockGetContentItemById.mockRejectedValue({ message: 'Mock Error' });

    const wrapper = shallow<Visualization>(<Visualization stagingEnvironment="vse" contentItemId="content" />, {
      lifecycleExperimental: true
    });

    expect(toJson(wrapper)).toMatchSnapshot();

    await waitUntil(() => wrapper.state('error') !== undefined);
    expect(mockGetContentItemById).toHaveBeenCalled();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should only load content when props are provided', async (): Promise<void> => {
    mockGetContentItemById.mockResolvedValue({
      text: '# Markdown Text'
    });

    const wrapper = shallow<Visualization>(<Visualization stagingEnvironment="" contentItemId="" />, {
      lifecycleExperimental: true
    });

    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.setProps({
      stagingEnvironment: 'vse',
      contentItemId: 'content'
    });

    await waitUntil(() => !wrapper.contains(<PageLoader />));

    expect(mockGetContentItemById).toHaveBeenCalled();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render a blog post', async () => {
    await renderVisualization(blogPostFixture);
  });

  it('should render a blog page', async () => {
    await renderVisualization(blogFixture as Blog);
    expect(findResultsState as jest.Mock).toBeCalled();
  });
});
