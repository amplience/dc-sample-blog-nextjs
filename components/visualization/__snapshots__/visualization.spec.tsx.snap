// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visualization should handle vse errors 1`] = `<PageLoader />`;

exports[`Visualization should handle vse errors 2`] = `
<Fragment>
  <h1>
    Unable to Render Visualization
  </h1>
  <p>
    Reason: 
    Mock Error
  </p>
</Fragment>
`;

exports[`Visualization should only load content when props are provided 1`] = `<PageLoader />`;

exports[`Visualization should only load content when props are provided 2`] = `
<Content
  content={
    Array [
      Object {
        "text": "# Markdown Text",
      },
    ]
  }
/>
`;

exports[`Visualization should render a blog page 1`] = `<PageLoader />`;

exports[`Visualization should render a blog page 2`] = `
<Fragment>
  <InstantSearch
    indexName="algolia-index-name"
    refresh={false}
    searchClient="mockSearchClient"
    stalledSearchDelay={200}
  >
    <ConnectorWrapper
      hitsPerPage={10}
    />
    <HeroBanner
      heading="heading"
    >
      <HeaderSearchBox
        placeholderText="text"
      />
    </HeroBanner>
    <Component />
    <SearchResultPagination />
  </InstantSearch>
</Fragment>
`;

exports[`Visualization should render a blog post 1`] = `<PageLoader />`;

exports[`Visualization should render a blog post 2`] = `
<Blog
  blogPost={
    Object {
      "_meta": Object {
        "deliveryId": "9cb44d61-d00b-49b7-b23b-72bb9a0a1de6",
        "deliveryKey": "blog",
        "name": "blog",
        "schema": "https://raw.githubusercontent.com/amplience/dc-sample-blog-nextjs/master/schemas/blog.json",
      },
      "authors": Array [
        Object {
          "_meta": Object {
            "deliveryId": "b1379f67-8934-49cb-94b3-58b679096c2c",
            "name": "bloblog-author",
            "schema": "https://schema.localhost.com/author.json",
          },
          "avatar": Object {
            "_meta": Object {
              "deliveryId": "9ef5a6d0-5a34-414f-9fd3-1098698c6629",
              "name": "bloblog-avatar",
              "schema": "https://schema.localhost.com/image.json",
            },
            "altText": "The dashing Sir Bloblog avatar",
            "image": Object {
              "_meta": Object {
                "schema": "http://bigcontent.io/cms/schema/v1/core#/definitions/image-link",
              },
              "defaultHost": "1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io",
              "endpoint": "bloblogltd",
              "id": "f9549d3d-f616-487c-b1ad-fad5152f2c7d",
              "mediaType": "i",
              "name": "man-business-suit",
            },
            "src": "//1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io/i/bloblogltd/man-business-suit",
          },
          "name": "Sir Bloblog",
        },
      ],
      "content": Array [
        Object {
          "_meta": Object {
            "deliveryId": "a61f458e-7c22-4b5b-9b3d-962df3cbcdcd",
            "name": "casual-crotch-shot",
            "schema": "https://schema.localhost.com/image.json",
          },
          "altText": "Casual crotch shot",
          "image": Object {
            "_meta": Object {
              "schema": "http://bigcontent.io/cms/schema/v1/core#/definitions/image-link",
            },
            "defaultHost": "1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io",
            "endpoint": "bloblogltd",
            "id": "e1b511d2-1a33-47e7-8dc7-f460534cb0c7",
            "mediaType": "i",
            "name": "casual-wear",
          },
          "src": "//1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io/i/bloblogltd/casual-wear",
        },
        Object {
          "_meta": Object {
            "deliveryId": "f08432cd-525f-4e50-ab34-3316081f48bd",
            "name": "my-first-bloblog-text",
            "schema": "https://schema.localhost.com/text.json",
          },
          "text": "# This is my first Bloblog

## Look-a-header 2

### Look-a-header 3

#### Look-a-header 4

##### Look-a-header 5

###### Look-a-header 6

_**This**_ _is_ _some_ ~~sample~~ **text**.

![alt text](https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png \\"Logo Title Text 1\\")

This is some more sample text with a [LINK](http://bbc.co.uk)

> Comments are cool!
> 
> See what I mean

Look some code:

\`\`\`javascript
const something = 'anything';
calls(something);

\`\`\`

inline \`code\` is awesome underlined

* * *

<h4>A title</h4>

A table too:

| Hello | world | table |
| --- | --- | --- |
| some | values | here |
| some | values | here |
| some | values | here |
| some | values | here |

Get listed

*   Item One
*   Item Two
*   Item Three",
        },
        Object {
          "_meta": Object {
            "deliveryId": "ad568b60-fd02-4302-9ccc-9e35dac24255",
            "name": "my-first-bloblog-video",
            "schema": "https://schema.localhost.com/video.json",
          },
          "srcSet": Array [
            "https://i1-qa.adis.ws/v/bloblogltd/SampleVideo_1280x720_5mb/mp4_240p?protocol=https",
          ],
          "video": Object {
            "_meta": Object {
              "schema": "http://bigcontent.io/cms/schema/v1/core#/definitions/video-link",
            },
            "defaultHost": "1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io",
            "endpoint": "bloblogltd",
            "id": "721044de-d125-4a1a-8ddc-2201b9463f2d",
            "mediaType": "v",
            "name": "SampleVideo_1280x720_5mb",
          },
        },
      ],
      "date": "August 07, 2019",
      "description": "Some text for the description that explains this bloblog post",
      "id": "9cb44d61-d00b-49b7-b23b-72bb9a0a1de6",
      "image": Object {
        "_meta": Object {
          "deliveryId": "4b44e870-bf71-43f3-81b3-c46c32f4cde5",
          "name": "crotchy-bag-image",
          "schema": "https://schema.localhost.com/image.json",
        },
        "altText": "crotch bags",
        "image": Object {
          "_meta": Object {
            "schema": "http://bigcontent.io/cms/schema/v1/core#/definitions/image-link",
          },
          "defaultHost": "1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io",
          "endpoint": "bloblogltd",
          "id": "edc77f02-0e92-4b1a-88f9-711cacb5d650",
          "mediaType": "i",
          "name": "friends-shopping-bags",
        },
        "src": "//1hgphw4170lum1o2yx0py00raf.staging.bigcontent.io/i/bloblogltd/friends-shopping-bags",
      },
      "readTime": 12,
      "tags": Array [
        "sir bloblog",
        "you're it",
        "No tiggy backs",
      ],
      "title": "My First Bloblog Post",
    }
  }
/>
`;

exports[`Visualization should render image 1`] = `<PageLoader />`;

exports[`Visualization should render image 2`] = `
<Content
  content={
    Array [
      Object {
        "altText": "altText",
        "image": Object {
          "defaultHost": "i1-qa.adis.ws",
          "endpoint": "bloblogltd",
          "id": "edc77f02-0e92-4b1a-88f9-711cacb5d650",
          "name": "friends-shopping-bags",
        },
      },
    ]
  }
/>
`;

exports[`Visualization should render text 1`] = `<PageLoader />`;

exports[`Visualization should render text 2`] = `
<Content
  content={
    Array [
      Object {
        "text": "# Markdown Text",
      },
    ]
  }
/>
`;

exports[`Visualization should render video 1`] = `<PageLoader />`;

exports[`Visualization should render video 2`] = `
<Content
  content={
    Array [
      Object {
        "srcSet": Array [
          "https://i1-qa.adis.ws/v/bloblogltd/SampleVideo_1280x720_5mb/mp4_240p?protocol=https",
        ],
        "video": Object {
          "defaultHost": "i1-qa.adis.ws",
          "endpoint": "bloblogltd",
          "id": "721044de-d125-4a1a-8ddc-2201b9463f2d",
          "name": "SampleVideo_1280x720_5mb",
        },
      },
    ]
  }
/>
`;
